monkey/monkey_island/cc/models/telemetries/telemetry.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import List
+
+from mongoengine import DateTimeField, Document, DynamicField, EmbeddedDocumentField, StringField
+
+from monkey_island.cc.database import mongo
+from monkey_island.cc.models import CommandControlChannel
+from monkey_island.cc.models.utils import document_encryptor
+from monkey_island.cc.models.utils.document_encryptor import FieldNotFoundError, SensitiveField
+from monkey_island.cc.models.utils.field_encryptors.mimikatz_results_encryptor import (
+    MimikatzResultsEncryptor,
+)
+
+sensitive_fields = [
+    SensitiveField("data.credentials", MimikatzResultsEncryptor),
+    SensitiveField("data.mimikatz", MimikatzResultsEncryptor),
+]
+
+
+class Telemetry(Document):
+
+    data = DynamicField(required=True)
+    timestamp = DateTimeField(required=True)
+    monkey_guid = StringField(required=True)
+    telem_category = StringField(required=True)
+    command_control_channel = EmbeddedDocumentField(CommandControlChannel)
+
+    meta = {"strict": False}
+
+    @staticmethod
+    def save_telemetry(telemetry_dict: dict):
+        try:
+            telemetry_dict = document_encryptor.encrypt(sensitive_fields, telemetry_dict)
+        except FieldNotFoundError:
+            pass  # Not all telemetries require encryption
+
+        cc_channel = CommandControlChannel(
+            src=telemetry_dict["command_control_channel"]["src"],
+            dst=telemetry_dict["command_control_channel"]["dst"],
+        )
+        Telemetry(
+            data=telemetry_dict["data"],
+            timestamp=telemetry_dict["timestamp"],
+            monkey_guid=telemetry_dict["monkey_guid"],
+            telem_category=telemetry_dict["telem_category"],
+            command_control_channel=cc_channel,
+        ).save()
+
+    @staticmethod
+    def get_telemetry_by_query(query: dict, output_fields=None) -> List[dict]:
+        telemetries = mongo.db.telemetry.find(query, output_fields)
+        decrypted_list = []
+        for telemetry in telemetries:
+            try:
+                decrypted_list.append(document_encryptor.decrypt(sensitive_fields, telemetry))
+            except FieldNotFoundError:
+                decrypted_list.append(telemetry)
+        return decrypted_list
```

Comment:
Agreed, since we ended up not being able to proxy the actual model fields through encryption. I've created a new `telemetry_dal.py` to hopefully further emphasize the interface